chia/wallet/wallet_state_manager.py

```diff
@@ -1474,32 +1474,30 @@ async def coin_added(
             wallet_id,
             wallet_type,
         )
-        farmer_reward = False
-        pool_reward = False
-        if self.is_farmer_reward(height, coin):
-            farmer_reward = True
-        elif self.is_pool_reward(height, coin):
-            pool_reward = True
-
-        farm_reward = False
-        parent_coin_record: Optional[WalletCoinRecord] = await self.coin_store.get_coin_record(coin.parent_coin_info)
-        if parent_coin_record is not None and wallet_type.value == parent_coin_record.wallet_type:
-            change = True
+
+        if self.is_pool_reward(height, coin):
+            tx_type = TransactionType.COINBASE_REWARD
+        elif self.is_farmer_reward(height, coin):
+            tx_type = TransactionType.FEE_REWARD
         else:
-            change = False
+            tx_type = TransactionType.INCOMING_TX
 
-        if farmer_reward or pool_reward:
-            farm_reward = True
-            if pool_reward:
-                tx_type: int = TransactionType.COINBASE_REWARD.value
-            else:
-                tx_type = TransactionType.FEE_REWARD.value
-            timestamp = await self.wallet_node.get_timestamp_for_height(height)
+        coinbase = tx_type != TransactionType.INCOMING_TX
+        coin_confirmed_transaction = False
+        if not coinbase:
+            for record in all_unconfirmed_transaction_records:
+                if coin in record.additions and not record.confirmed:
+                    await self.tx_store.set_confirmed(record.name, height)
+                    coin_confirmed_transaction = True
```

Comment:
at this point, we won't ever find any other record with this same coin, will we?
it seems to me that we could `break` the for loop here

Reply:
Yeah right we probably won't unless we do something strange with the unconfirmed transactions i don't know about. Added https://github.com/Chia-Network/chia-blockchain/pull/14852/commits/8f1d4f066c33ec7231117adbb79382bf489a724a.